src/Connector.sol

```diff
@@ -81,15 +83,11 @@ contract CentrifugeConnector {
         emit File(what, data);
     }
 
-    function file(bytes32 name, string memory domainName, uint32 domainId) public auth  {
-        if(name == "domain") {
-           domainLookup[keccak256(bytes(domainName))] = domainId;
-           emit File(name, domainName);
-        } else { revert ("unknown name");}
-        
+    // --- Internal ---
+    function getDomain(Domain domain, uint domainId) public pure returns (bytes32) {
+        return bytes32(uint64(domain) + domainId);
```

Comment:
I like it! Gonna implement your suggestion

Reply:
I just realised, to your comment earlier @NunoAlexandre that Moonbeam actually is the EVM itself, so it would be identified by the EVM domain with the EVM chain ID. And given that we're quite unlikely to implement Connectors in anything other than Solidity / targeting the EVM for quite a while. Maybe we should just simplify this to

```
enum Domain { EVM, Centrifuge }
```

And make `domainId` optional. That way if the target is an EVM, then we use chain IDs, and if the target is Centrifuge, there's no additional ID required. In the far future when we want to implement non-EVM targets, we can re-add a `Parachain` domain, but it still makes sense to have `Centrifuge` separate as it encodes more efficiently.

Reply:
@offerijns cool, this starts making more conceptual sense to me. However, how do we map this logic into the cent-chain, specially when it comes to special cases like Moonbeam? I guess we would specify "Moonbeam" by specifying its EVM Chain ID and we would internally look up the Router associated with that domain?